File: Programmers/타겟넘버.py
def solution(numbers, target):
    from collections import deque
    que = deque()
    que.append(0)

    for i in range(len(numbers)):
        lenth = len(que)
        for j in range(lenth):
            x = que.popleft()
            que.append(x + numbers[i])
            que.append(x - numbers[i])

    answer = 0
    for x in que:
        if x == target: answer += 1
    return answer

File: Programmers/타겟넘버.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

int solution(vector<int> numbers, int target) {
    int answer = 0;
    queue<int> que;

    que.push(0);
    for (int i=0; i<numbers.size(); i++) {
        int lent = que.size();
        while (lent--) {
            int x = que.front();
            que.pop();
            que.push(x + numbers[i]);
            que.push(x - numbers[i]);
        }
    }
    int q_lenth = que.size();
    for (int i=0; i<q_lenth; i++) {
        int x = que.front();
        que.pop();
        if (x == target) {
            answer++;
        }
    }
    return answer;
}